tests/monotouch-test/ImageIO/CGImageAnimationTest.cs

```diff
@@ -0,0 +1,186 @@
+
+			bool done = false;
+			TestRuntime.RunAsync (TimeSpan.FromSeconds (30), async () => {
+				imageAnimation.AnimateImage(imageUrl, null, MyBlockSetValueZero);
+				await taskCompletionSource.Task;
+				done = true;
+			},
+				() => done);
+			Assert.AreEqual (0, testValue);
+		}
+
+		[Test]
+		public void AnimateImageWithData ()
+		{
+
+			Assert.AreEqual (-1, testValue);
+			TaskCompletionSource<bool> taskCompletionSource = new TaskCompletionSource<bool> ();
+
+			bool done = false;
+			TestRuntime.RunAsync (TimeSpan.FromSeconds (30), async () => {
+				imageAnimation.AnimateImage(imageData, null, MyBlockSetValueZero);
+				await taskCompletionSource.Task;
+				done = true;
+			},
+				() => done);
+			Assert.AreEqual (0, testValue);
+		}
+
+		[Test]
+		public void AnimateImageWithUrlChangeBlock ()
+		{
+			Assert.AreEqual (-1, testValue);
+			TaskCompletionSource<bool> taskCompletionSource = new TaskCompletionSource<bool> ();
+
+			bool done = false;
+			TestRuntime.RunAsync (TimeSpan.FromSeconds (30), async () => {
+				imageAnimation.AnimateImage (imageUrl, null, MyBlockSetValueZero);
+				await taskCompletionSource.Task;
+				done = true;
+			},
+				() => done);
+			Assert.AreEqual (0, testValue);
+
+			done = false; // reset after first call
+			TestRuntime.RunAsync (TimeSpan.FromSeconds (30), async () => {
+				imageAnimation.AnimateImage (imageUrl, null, MyBlockSetValueOne);
+				await taskCompletionSource.Task;
+				done = true;
+			},
+				() => done);
+			Assert.AreEqual (1, testValue);
+		}
+
+		[Test]
+		public void AnimateImageWithDataChangeBlock ()
+		{
+			Assert.AreEqual (-1, testValue);
+			TaskCompletionSource<bool> taskCompletionSource = new TaskCompletionSource<bool> ();
+
+			bool done = false;
+			TestRuntime.RunAsync (TimeSpan.FromSeconds (30), async () => {
+				imageAnimation.AnimateImage (imageData, null, MyBlockSetValueZero);
+				await taskCompletionSource.Task;
+				done = true;
+			},
+				() => done);
+			Assert.AreEqual (0, testValue);
+
+			done = false; // reset after first call
+			TestRuntime.RunAsync (TimeSpan.FromSeconds (30), async () => {
+				imageAnimation.AnimateImage (imageData, null, MyBlockSetValueOne);
+				await taskCompletionSource.Task;
+				done = true;
+			},
+				() => done);
+			Assert.AreEqual (1, testValue);
+		}
+
+		[Test]
+		public void AnimateImageWithUrlNullUrl()
+		{
+			NSUrl nullUrl = null;
+			Assert.Throws<ArgumentNullException>(() => imageAnimation.AnimateImage(nullUrl, null, MyBlockSetValueZero));
+		}
+
+		[Test]
+		public void AnimateImageWithUrlNullBlock()
+		{
+			Assert.Throws<ArgumentNullException>(() => imageAnimation.AnimateImage(imageUrl, null, /* CGImageSourceAnimationBlock */ null));
+		}
+
+		[Test]
+		public void AnimateImageWithDataNullData()
+		{
+			NSData nullData = null;
+			Assert.Throws<ArgumentNullException>(() => imageAnimation.AnimateImage(nullData, null, MyBlockSetValueZero));
```

Comment:
minor: missing space before `(`

Reply:
minor: you do not need a local variable, you can just cast the null, e.g. `(NSData)null`

Reply:
fixed 👍